<a href="https://colab.research.google.com/github/Marcelo-jmd/PracticandoPy/blob/main/Practicando_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<h1>Recopilación y muestra de datos<h1>
<p>1 - Crea un programa que solicite al usuario que escriba su nombre y luego imprima "Hola, [nombre]."<p>


In [ ]:
nombre = input('Ingrese su nombre: ')
print(f'Hola {nombre}')

Ingrese su nombre: Marcelo
Hola Marcelo


2 - Crea un programa que solicite al usuario que escriba su nombre y edad, y luego imprima "Hola, [nombre], tienes [edad] años."

In [ ]:
nom = input('Ingrese su nombre: ')
edad = input('Ingrese su edad: ')
print('Hola, {}, tienes {} años'.format(nom,edad))

Ingrese su nombre: Marcelo
Ingrese su edad: 23
Hola, Marcelo, tienes 23 años


3 - Crea un programa que solicite al usuario que escriba su nombre, edad y altura en metros, y luego imprima "Hola, [nombre], tienes [edad] años y mides [altura] metros."


In [ ]:
nom = input('Ingrese su nombre: ')
edad = int(input('Ingrese su edad: '))
alt = float(input('Ingrese su altura: '))
print('Hola, {}, tienes {} años y mides {} metros'.format(nom,edad,alt))

Ingrese su nombre: Marcelo
Ingrese su edad: 23
Ingrese su altura: 1.67
Hola, Marcelo, tienes 23 años y mides 1.67 metros


# Calculadora con operadores

4 - Crea un programa que solicite dos valores numéricos al usuario y luego imprima la suma de ambos valores.

In [ ]:
var1 = int(input('Primer número: '))
var2 = int(input('Segundo número: '))
print('La suma de {} y {} es: {}'.format(var1,var2,var1+var2))

Primer número: 10
Segundo número: 15
La suma de 10 y 15 es: 25


5 - Crea un programa que solicite tres valores numéricos al usuario y luego imprima la suma de los tres valores.


In [ ]:
a = int(input('1er número: '))
b = int(input('2do número: '))
c = int(input('3er número: '))
sum = a+b+c
print('La suma de {},{} y {} es: {}'.format(a,b,c,sum))

1er número: 1
2do número: 2
3er número: 3
La suma de 1,2 y 3 es: 6


6 - Crea un programa que solicite dos valores numéricos al usuario y luego imprima la resta del primero menos el segundo valor.


In [ ]:
a = int(input('1er número: '))
b = int(input('2do número: '))
dif = a-b
print('La resta de {} con {} es: {}'.format(a,b,dif))

1er número: 7
2do número: 5
La resta de 7 con 5 es: 2


7 - Crea un programa que solicite dos valores numéricos al usuario y luego imprima la multiplicación de los dos valores.

In [ ]:
a = int(input('1er número: '))
b = int(input('2do número: '))
pro = a*b
print('El producto de {} con {} es: {}'.format(a,b,pro))

1er número: 5
2do número: 3
El producto de 5 con 3 es: 15


8 - Crea un programa que solicite dos valores numéricos, un numerador y un denominador, y realice la división entre los dos valores. Asegúrate de que el valor del denominador no sea igual a 0.

In [ ]:
a = int(input('1er número: '))
b = int(input('2do número: '))
if b != 0:
  div = a/b
else:
  div = 0
print('La división de {} con {} es: {}'.format(a,b,div))

1er número: 6
2do número: 0
La división de 6 con 0 es: 0


9 - Crea un programa que solicite dos valores numéricos, un operador y una potencia, y realice la exponenciación entre estos dos valores.


In [ ]:
a = int(input('1er número: '))
b = int(input('2do número: '))
pot = a**b
print('La potencia de {} a la {} es: {}'.format(a,b,pot))

1er número: 2
2do número: 3
La potencia de 2 a la 3 es: 8


10 - Crea un programa que solicite dos valores numéricos, un numerador y un denominador, y realice la división entera entre los dos valores. Asegúrate de que el valor del denominador no sea igual a 0.


In [ ]:
a = int(input('1er número: '))
b = int(input('2do número: '))
if b != 0:
  div = a/b
else:
  div = 0
print('La división de {} con {} es: {}'.format(a,b,div))

1er número: 6
2do número: 0
La división de 6 con 0 es: 0


11 - Crea un programa que solicite dos valores numéricos, un numerador y un denominador, y devuelva el resto de la división entre los dos valores. Asegúrate de que el valor del denominador no sea igual a 0.

In [ ]:
a = int(input('1er número: '))
b = int(input('2do número: '))
if b != 0:
  div = a%b
else:
  div = 0
print('El resto de {} con {} es: {}'.format(a,b,div))

1er número: 12
2do número: 4
El resto de 12 con 4 es: 0


12 - Crea un código que solicite las 3 notas de un estudiante e imprima el promedio de las notas.

In [ ]:
a = int(input('1er nota: '))
b = int(input('2do nota: '))
c = int(input('3er nota: '))
prom = (a+b+c)/3
print('El promedio de los números es: %d'%(prom))

1er nota: 5
2do nota: 3
3er nota: 1
El promedio de los números es: 3


13 - Crea un código que calcule e imprima el promedio ponderado de los números 5, 12, 20 y 15 con pesos respectivamente iguales a 1, 2, 3 y 4.

In [ ]:
num1 = 5
pes1 = 1
num2 = 12
pes2 = 2
num3 = 20
pes3 = 3
num4 = 15
pes4 = 4
prom_ponderado = (num1*pes1+num2*pes2+num3*pes3+num4*pes4)/(num1+num2+num3+num4)
print('El promedio ponderado es: %.2f'%(prom_ponderado))

El promedio ponderado es: 2.87


# Editando textos

14 - Crea una variable llamada "frase" y asígnale una cadena de texto de tu elección. Luego, imprime la frase en pantalla.


In [ ]:
frase = 'Hola tengo sueño'
print(f'Esta es mi frase "{frase}"')

Esta es mi frase "Hola tengo sueño"


15 - Crea un código que solicite una frase y luego imprima la frase en pantalla.

In [ ]:
frase = input('Ingresa una frase:')
print(f'Esta es mi frase "{frase}"')

Ingresa una frase:Quiero comer
Esta es mi frase "Quiero comer"


16 - Crea un código que solicite una frase al usuario y luego imprima la misma frase ingresada pero en mayúsculas.


In [ ]:
frase = input('Ingresa una frase:')
print(f'Esta es tu frase "{frase.upper()}"')

Ingresa una frase:Soy Iroman
Esta es tu frase "SOY IROMAN"


17 - Crea un código que solicite una frase al usuario y luego imprima la misma frase ingresada pero en minúsculas.

In [ ]:
frase = input('Ingresa una frase:')
print(f'Esta es tu frase "{frase.lower()}"')

Ingresa una frase:Soy peter parker
Esta es tu frase "soy peter parker"


18 - Crea una variable llamada "frase" y asígnale una cadena de texto de tu elección. Luego, imprime la frase sin espacios en blanco al principio y al final.

In [ ]:
frase = '  Hola tengo sueño '
print(f'Esta es mi frase "{frase.strip()}"')

Esta es mi frase "Hola tengo sueño"


19 - Crea un código que solicite una frase al usuario y luego imprima la misma frase sin espacios en blanco al principio y al final.


In [ ]:
frase = input('Ingresa tu frase: ')
print(f'Esta es tu frase "{frase.strip()}"')

Ingresa tu frase:   La chica de dollarcity xdd 
Esta es tu frase "La chica de dollarcity xdd"


20 - Crea un código que solicite una frase al usuario y luego imprima la misma frase sin espacios en blanco al principio y al final, además de convertirla a minúsculas.


In [ ]:
frase = input('Ingresa tu frase: ')
print(f'Esta es tu frase "{frase.strip().lower()}"')

Ingresa tu frase:    Hakuna Matata 
Esta es tu frase "hakuna matata"


21 - Crea un código que solicite una frase al usuario y luego imprima la misma frase con todas las vocales "e" reemplazadas por la letra "f".

In [ ]:
frase = input('Ingresa tu frase: ')
print('Esta es tu frase "{}"'.format(frase.replace('e','f')))

Ingresa tu frase: El perro trae el omitrix
Esta es tu frase "El pfrro traf fl omitrix"


22 - Crea un código que solicite una frase al usuario y luego imprima la misma frase con todas las vocales "a" reemplazadas por el carácter "@".


In [ ]:
frase = input('Ingresa tu frase: ')
print('Esta es tu frase "{}"'.format(frase.replace('a','@')))

Ingresa tu frase: Quiero comer pizzaa
Esta es tu frase "Quiero comer pizz@@"


23 - Crea un código que solicite una frase al usuario y luego imprima la misma frase con todas las consonantes "s" reemplazadas por el carácter "$".

In [ ]:
frase = input('Ingresa tu frase: ')
print('Esta es tu frase "{}"'.format(frase.replace('s','$')))

Ingresa tu frase: Casi ingreso una frase sin S
Esta es tu frase "Ca$i ingre$o una fra$e $in S"


# Entrenando la programación

1 - Escribe un programa que pida a la persona usuaria que proporcione dos números y muestre el número más grande.

In [ ]:
a = int(input('1er número: '))
b = int(input('2do número: '))
if a >= b:
  print('El número mayor es: {}'.format(a))
else:
  print('El número mayor es: {}'.format(b))

1er número: 20
2do número: 3
El número mayor es: 20


2 - Escribe un programa que solicite el porcentaje de crecimiento de producción de una empresa e informe si hubo un crecimiento (porcentaje positivo) o una disminución (porcentaje negativo).

In [ ]:
x = float(input('Ingrese porcentaje de crecimietn(%): '))
if x >= 0:
  print('Hubo un CRECIMIENTO DE PRODUCCCIÓN')
else:
  print('Hubo una DISMINUCIÓN DE PRODUCCCIÓN')

Ingrese porcentaje de crecimietn(%): -9.2
Hubo una DISMINUCIÓN DE PRODUCCCIÓN


3 - Escribe un programa que determine si una letra proporcionada por la persona usuaria es una vocal o una consonante.

In [ ]:
x = input('Ingrese una letra: ')
if x == 'a' or x == 'e' or x=='i' or x == 'o' or x == 'u':
  print('La letra {} es una VOCAL'.format(x))
else:
  print('La letra {} es una CONSONANTE'.format(x))

Ingrese una letra: w
La letra w es una CONSONANTE


4 - Escribe un programa que lea valores promedio de precios de un modelo de automóvil durante 3 años consecutivos y muestre el valor más alto y más bajo entre esos tres años.

In [ ]:
p1 = int(input('1er número: '))
p2 = int(input('2do número: '))
p3 = int(input('3er número: '))
mayor = p1
menor = 0
if p2 >= p1 and p2 >= p3:
  mayor = p2
  if p1 >= p3:
    menor = p3
  else:
    menor = p1
elif p3>= p1 and p3 >= p2:
  mayor = p3
  if p2 >= p1:
    menor = p1
  else:
    menor = p2
print('El promedio mayor es: {}'.format(mayor))
print('El promedio menor es: {}'.format(menor))

1er número: 351
2do número: 352
3er número: 52
El promedio mayor es: 352
El promedio menor es: 52


5 - Escribe un programa que pregunte sobre el precio de tres productos e indique cuál es el producto más barato para comprar.

In [ ]:
p1 = int(input('1er precio: '))
p2 = int(input('2do precio: '))
p3 = int(input('3er precio: '))
menor = p1
if p2 <= p1 and p2 <= p3:
  menor = p2
elif p3<= p1 and p3 <= p2:
  menor = p3

print('El precio menor es: {}'.format(menor))

1er precio: 51
2do precio: 5
3er precio: 512
El precio menor es: 5


6 - Escribe un programa que lea tres números y los muestre en orden descendente.

In [ ]:
p1 = int(input('1er número: '))
p2 = int(input('2do número: '))
p3 = int(input('3er número: '))

if p1 >= p2 and p1 >= p3:
  mayor = p1
  if p2 >= p3:
    medio = p2
    menor = p3
  else:
    medio = p3
    menor = p2
elif p2 >= p1 and p2 >= p3:
  mayor = p2
  if p1>= p3:
    medio = p1
    menor = p3
  else:
    medio = p3
    menor = p1
else:
  mayor = p3
  if p1>= p2:
    medio = p1
    menor = p2
  else:
    medio = p2
    menor = p1

print('Los números ordenados en orden descendente: {}, {}, {}'.format(mayor,medio,menor))

1er número: 85
2do número: 43
3er número: 765
Los números ordenados en orden descendente: 765, 85, 43


7 -Escribe un programa que pregunte en qué turno estudia la persona usuaria ("mañana", "tarde" o "noche") y muestre el mensaje "¡Buenos Días!", "¡Buenas Tardes!", "¡Buenas Noches!" o "Valor Inválido!", según el caso.

In [ ]:
turno = input('Ingrese turno: ')
if turno == 'mañana':
  print('Buenos Días!')
elif turno == 'tarde':
  print('Buenos Tardes!')
elif turno == 'noche':
  print('Buenos Noches!')
else:
  print('Valor Inválido')

Ingrese turno: tarde
Buenos Tardes!


8 - Escribe un programa que solicite un número entero a la persona usuaria y determine si es par o impar. Pista: Puedes usar el operador módulo (%).

In [ ]:
num = int(input('Ingrese un número: '))
if num%2 == 0:
  print('El número es par')
else:
  print('El número es impar')

Ingrese un número: 9
El número es impar


9 - Escribe un programa que pida un número a la persona usuaria y le informe si es entero o decimal.

In [ ]:
print(int(5.2))

5


In [ ]:
num = float(input('Ingrese un número: '))
if num== int(num):
    print("Es un número entero.")
else:
    print("Es un número decimal.")

Ingrese un número: 6
Es un número entero.


# Momento de los proyectos

10 - Un programa debe ser escrito para leer dos números y luego preguntar a la persona usuaria qué operación desea realizar. El resultado de la operación debe incluir información sobre el número, si es par o impar, positivo o negativo, e entero o decimal.

In [ ]:
n1 = float(input('Ingresa 1er número: '))
n2 = float(input('Ingresa 2do número: '))
op = input('Ingrese operación a realizar (s,r,m,d): ')
if op =='s':
  res = n1+n2
elif op == 'r':
  res = n1-n2
elif op == 'm':
  res = n1*n2
else:
  res = n1/n2
print(f'El resultado es {res}')
if res%2 == 0:
  print('El resultado es PAR')
else:
  print('El resultado es IMPAR')
if res >= 0:
  print('El resultado es POSITIVO')
else:
  print('El resultado es NEGATIVO')
if res == int(res):
  print('El resultado es entero')
else:
  print('El resultado es decimal')

Ingresa 1er número: 5
Ingresa 2do número: 1.5
Ingrese operación a realizar (s,r,m,d): m
El resultado es 7.5
El resultado es IMPAR
El resultado es POSITIVO
El resultado es decimal


11 - Escribe un programa que pida a la persona usuaria tres números que representan los lados de un triángulo. El programa debe informar si los valores pueden utilizarse para formar un triángulo y, en caso afirmativo, si es equilátero, isósceles o escaleno. Ten en cuenta algunas sugerencias:

Tres lados forman un triángulo cuando la suma de cualesquiera dos lados es mayor que el tercero;
Triángulo Equilátero: tres lados iguales;
Triángulo Isósceles: dos lados iguales;
Triángulo Escaleno: tres lados diferentes.

In [ ]:
a = int(input('Ingrese 1er lado: '))
b = int(input('Ingrese 2do lado: '))
c = int(input('Ingrese 3er lado: '))
if a<b+c or b<a+c or c<a+b:
  if a == b and b == c and c == a:
    tip = 'Equilátero'
  elif a == b or b == c or a == c:
    tip = 'Isósceles'
  else:
    tip = 'Escaleno'
  print(f'El triángulo es de tipo {tip}')
else:
  print('No se puede formar un triángulo')


Ingrese 1er lado: 0
Ingrese 2do lado: 0
Ingrese 3er lado: 0
No se puede formar un triángulo


12 - Un establecimiento está vendiendo combustibles con descuentos variables. Para el etanol, si la cantidad comprada es de hasta 15 litros, el descuento será del 2% por litro. En caso contrario, será del 4% por litro. Para el diésel, si la cantidad comprada es de hasta 15 litros, el descuento será del 3% por litro. En caso contrario, será del 5% por litro. El precio por litro de diésel es de R$ 2,00 y el precio por litro de etanol es de R$ 1,70. Escribe un programa que lea la cantidad de litros vendidos y el tipo de combustible (E para etanol y D para diésel) y calcule el valor a pagar por el cliente. Ten en cuenta algunas sugerencias:

El valor del descuento será el producto del precio por litro, la cantidad de litros y el valor del descuento.
El valor a pagar por un cliente será el resultado de la multiplicación del precio por litro por la cantidad de litros menos el valor del descuento resultante del cálculo.

In [ ]:
l = float(input('Ingrese litros: '))
tip = input('Ingrese tipo de combustible (E/D):')
if tip == 'E':
  prec = 2
  if l <= 15:
    desc = 0.02
  else:
    desc = 0.04

elif tip == 'D':
  prec = 1.7
  if l <= 15:
    desc = 0.03
  else:
    desc = 0.05
else:
  print('Tipo de combustible inválido')
pagar = prec*l-prec*l*desc
print('Usted compro {} litros de {} con un descuento de {} %, total a pagar: {}'.format(l,tip,desc*100,pagar))

Ingrese litros: 30
Ingrese tipo de combustible (E/D):D
Usted compro 30.0 litros de D con un descuento de 5.0 %, total a pagar: 48.45


13 - En una empresa de venta de bienes raíces, debes crear un código que analice los datos de ventas anuales para ayudar a la dirección en la toma de decisiones. El código debe recopilar los datos de cantidad de ventas durante los años 2022 y 2023 y calcular la variación porcentual. A partir del valor de la variación, se deben proporcionar las siguientes sugerencias:

Para una variación superior al 20%: bonificación para el equipo de ventas.
Para una variación entre el 2% y el 20%: pequeña bonificación para el equipo de ventas.
Para una variación entre el 2% y el -10%: planificación de políticas de incentivo a las ventas.
Para bonificaciones inferiores al -10%: recorte de gastos.

In [ ]:
year1 = int(input('Venta anual del 2022: '))
year2 = int(input('Venta anual del 2023: '))
var = ((year2-year1)/year1)*100
if var > 20:
  est = 'Bonificación'
elif var > 2:
  est = 'Pequeña bonificación'
elif var > -10:
  est = 'Planificación de políticas de incentivo'
else:
  est = 'Recorte de gastos'
print('El estado del equipo de ventas es: '+est)

Venta anual del 2022: 100
Venta anual del 2023: 99
El estado del equipo de ventas es: Planificación de políticas de incentivo


# Entrenando la programación

1 - Escribe un programa que solicite dos números enteros e imprima todos los números enteros entre ellos.

2 - Escribe un programa para calcular cuántos días tomará que la colonia de una bacteria A supere o iguale a la colonia de una bacteria B, basado en tasas de crecimiento del 3% y 1.5%, respectivamente. Supón que la colonia A comienza con 4 elementos y B con 10.

3 - Para procesar una cantidad de 15 datos de evaluaciones de usuarios de un servicio de la empresa, necesitamos verificar si las calificaciones son válidas. Por lo tanto, escribe un programa que recibirá calificaciones del 0 al 5 y verificará si son valores válidos. Si se ingresa una calificación superior a 5 o inferior a 0, se repetirá hasta que el usuario ingrese un valor válido.

4 - Desarrolla un programa que lea un conjunto indefinido de temperaturas en grados Celsius y calcule su promedio. La lectura debe detenerse al ingresar el valor -273°C.

5 - Escribe un programa que calcule el factorial de un número entero proporcionado por el usuario. Recuerda que el factorial de un número entero es el producto de ese número por todos sus antecesores hasta llegar al número 1. Por ejemplo, el factorial de 5 es 5 x 4 x 3 x 2 x 1 = 120.

# Momento de los proyectos

6 - Escribe un programa que genere la tabla de multiplicar de un número entero del 1 al 10, según la elección del usuario. Como ejemplo, para el número 2, la tabla de multiplicar debe mostrarse en el siguiente formato:

Tabla de multiplicar del 2:
2 x 1 = 2
2 x 2 = 4
[...]
2 x 10 = 20

7 - Los números primos tienen diversas aplicaciones en Ciencia de Datos, como en criptografía y seguridad. Un número primo es aquel que es divisible solo por sí mismo y por 1. Por lo tanto, crea un programa que solicite un número entero y determine si es un número primo o no.

8 - Vamos a comprender la distribución de edades de los pensionistas de una empresa de seguros. Escribe un programa que lea las edades de una cantidad no informada de clientes y muestre la distribución en los intervalos [0-25], [26-50], [51-75] y [76-100]. La entrada de datos se detendrá al ingresar un número negativo.

9 - En una elección para la gerencia de una empresa con 20 empleados, hay cuatro candidatos. Escribe un programa que calcule al ganador de la elección. La votación se realizó de la siguiente manera:

Cada empleado votó por uno de los cuatro candidatos (representados por los números 1, 2, 3 y 4).

También se contaron los votos nulos (representados por el número 5) y los votos en blanco (representados por el número 6).

Al final de la votación, el programa debe mostrar el número total de votos para cada candidato, los votos nulos y los votos en blanco. Además, debe calcular y mostrar el porcentaje de votos nulos con respecto al total de votos y el porcentaje de votos en blanco con respecto al total de votos.